File: backend/app/schemas/rbac.py
from datetime import datetime

from pydantic import BaseModel


class PolicyRule(BaseModel):
    api_groups: list[str]
    resources: list[str]
    verbs: list[str]


class ClusterRoleInfo(BaseModel):
    name: str
    rules_count: int
    rules: list[PolicyRule]
    created_at: datetime | None


class RoleSubject(BaseModel):
    kind: str
    name: str
    namespace: str | None


class ClusterRoleBindingInfo(BaseModel):
    name: str
    role_kind: str | None
    role_name: str | None
    subjects: list[RoleSubject]
    created_at: datetime | None


class ServiceAccountInfo(BaseModel):
    name: str
    namespace: str | None
    secrets_count: int
    created_at: datetime | None

File: frontend/src/types/rbac.ts
export interface PolicyRule {
  api_groups: string[];
  resources: string[];
  verbs: string[];
}

export interface ClusterRoleInfo {
  name: string;
  rules_count: number;
  rules: PolicyRule[];
  created_at: string | null;
}

export interface RoleSubject {
  kind: string;
  name: string;
  namespace: string | null;
}

export interface ClusterRoleBindingInfo {
  name: string;
  role_kind: string | null;
  role_name: string | null;
  subjects: RoleSubject[];
  created_at: string | null;
}

export interface ServiceAccountInfo {
  name: string;
  namespace: string | null;
  secrets_count: number;
  created_at: string | null;
}
